src/app/containers/MostRead/Canonical/index.jsx

```diff
@@ -56,18 +60,28 @@ const CanonicalMostRead = ({
 
   useEffect(() => {
     if (!items) {
-      const handleResponse = async response => {
+      const handleResponse = url => async response => {
         if (!response.ok) {
-          throw Error(response.statusText);
+          throw Error(
+            `Unexpected response (HTTP status code ${response.status}) when requesting ${url}`,
+          );
         }
         const mostReadData = await response.json();
         setItems(processMostRead({ data: mostReadData, numberOfItems }));
       };
 
-      const fetchMostReadData = pathname =>
-        fetch(pathname, { mode: 'no-cors' })
-          .then(handleResponse)
-          .catch(e => logger.error(`HTTP Error: "${e}"`));
+      const fetchMostReadData = pathname => {
+        logger.info(MOST_READ_CLIENT_REQUEST, { url: endpoint });
+
+        return fetch(pathname, { mode: 'no-cors' })
```

Comment:
What would happen if we remove no-cors mode?

Reply:
I read up on this and I _think_ no-cors means this can only be a GET or POST (or HEAD), I thought it also meant that javascript couldn't read the response, but it obviously can if we're doing it here. It might just be that service workers can't. @thekp did you add this originally or did someone else, I missed any conversations about this :D

Reply:
I implemented this originally, and I remember having problems fetching data without this mode. But just tested locally without the `{ mode: 'no-cors' }` and it seems to work on local dev and storybook.

So I shall remove it 👍

Reply:
I committed this in my recent commit (mindlessly did `git add .` and wasn't paying attention to what I was committing)